src/e2e-test/features/recordsplitter/BigQueryToRecordSplitter.feature

```diff
@@ -0,0 +1,160 @@
+    When Expand Plugin group in the LHS plugins list: "Transform"
+    When Select plugin: "Record Splitter" from the plugins list as: "Transform"
+    Then Connect plugins: "BigQuery" and "RecordSplitter" to establish connection
+    When Expand Plugin group in the LHS plugins list: "Sink"
+    When Select plugin: "GCS" from the plugins list as: "Sink"
+    Then Connect plugins: "RecordSplitter" and "GCS" to establish connection
+    Then Navigate to the properties page of plugin: "BigQuery"
+    Then Replace input plugin property: "project" with value: "projectId"
+    Then Enter input plugin property: "datasetProject" with value: "projectId"
+    Then Override Service account details if set in environment variables
+    Then Enter input plugin property: "referenceName" with value: "BQReferenceName"
+    Then Enter input plugin property: "dataset" with value: "dataset"
+    Then Enter input plugin property: "table" with value: "bqSourceTable"
+    Then Click on the Get Schema button
+    Then Capture the generated Output Schema
+    Then Validate "BigQuery" plugin properties
+    Then Close the Plugin Properties page
+    Then Navigate to the properties page of plugin: "RecordSplitter"
+    Then Enter input plugin property: "fieldToSplit" with value: "recordSplitterFieldToSplit"
+    Then Enter input plugin property: "delimiter" with value: "recordSplitterCommaDelimiter"
+    Then Enter input plugin property: "outputField" with value: "recordSplitterOutputField"
+    Then Select Record Splitter plugin output schema action: "clear"
+    Then Enter Record Splitter plugin outputSchema "recordSplitterValidOutputSchema"
+    Then Validate "RecordSplitter" plugin properties
+    Then Close the Plugin Properties page
+    Then Navigate to the properties page of plugin: "GCS"
+    Then Replace input plugin property: "project" with value: "projectId"
+    Then Override Service account details if set in environment variables
+    Then Enter input plugin property: "referenceName" with value: "GCSReferenceName"
+    Then Enter input plugin property: "path" with value: "gcsTargetPath"
+    Then Replace input plugin property: "suffix" with value: "yyyy-MM-dd-HH-mm-ss"
+    Then Select dropdown plugin property: "format" with option value: "csv"
+    Then Validate "GCS" plugin properties
+    Then Close the Plugin Properties page
+    Then Save the pipeline
+    Then Preview and run the pipeline
+    Then Wait till pipeline preview is in running state
+    Then Open and capture pipeline preview logs
+    Then Verify the preview run status of pipeline in the logs is "succeeded"
+    Then Close the pipeline logs
+    Then Close the preview
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
+    Then Close the pipeline logs
+    Then Validate OUT record count of record splitter is equal to IN record count of sink
```

Comment:
number of output records depends on the input data (delimiters in the field that's being split). We should validate that record splitter correctly split the input data. Checking records out of splitter == records in to sink just tells us that the pipeline ran successfully, doesn't validate that splitter functionality

Reply:
Added the validation step.